jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupServiceImpl.java

```diff
@@ -65,7 +67,14 @@ public GroupServiceImpl(final GroupPersistenceService groupPersistenceService,
     @Transactional
     public Group createGroup(final CreateGroupRequest createGroupRequest,
                              final User aCreatingUser) {
-
+        try {
+            groupPersistenceService.getGroup(createGroupRequest.getGroupName());
+            String message = MessageFormat.format("Group Name already exists: {0} ", createGroupRequest.getGroupName());
+            LOGGER.error(message);
+            throw new EntityExistsException(message);
+        } catch (NotFoundException e) {
+            LOGGER.debug("No group name conflict, ignoring not found exception for creating group ", e);
+        }
         createGroupRequest.validate();
```

Comment:
Does validate check if the group name is empty or has illegal characters ? If so, createGroupRequest.validate() should go first before you check for duplicates.